Tests/MongoSwiftTests/ChangeStreamTests.swift

```diff
@@ -14,153 +108,232 @@ final class ChangeStreamTest: MongoSwiftTestCase {
+        try coll.findOneAndDelete(["x": 2])
+        let change3 = changeStream.next()
+        expect(change3).toNot(beNil())
+        expect(change3?.operationType).to(equal(.delete))
+
+        // expect the resumeToken to be updated to the _id field of the most recently accessed document
+        expect(changeStream.resumeToken).to(equal(change3?._id))
+    }
+
+    func testChangeStreamWithPipeline() throws {
+        guard MongoSwiftTestCase.topologyType != .single else {
+            print("Skipping test case because of unsupported topology type \(MongoSwiftTestCase.topologyType)")
+            return
         }
+
+        let client = try MongoClient()
+        let db = client.db(type(of: self).testDatabase)
+        defer { try? db.drop() }
+        let coll = try db.createCollection(self.getCollectionName(suffix: "1"))
+        let options = ChangeStreamOptions(fullDocument: .updateLookup)
+        let pipeline: [Document] = [["$match": ["fullDocument.a": 1] as Document]]
+        let changeStream = try coll.watch(pipeline, options: options)
+
+        let doc1: Document = ["_id": 1, "a": 1]
+        try coll.insertOne(doc1)
+        let change1 = changeStream.next()
+
+        expect(change1).toNot(beNil())
+        expect(change1?.operationType).to(equal(.insert))
+        expect(change1?.fullDocument).to(equal(doc1))
+        // test that a change when inserting a field not included in the pipeline
+        try coll.insertOne(["b": 2])
+        let change2 = changeStream.next()
+        expect(change2).to(beNil())
     }
 
-    func testChangeStreamOnADatabase() throws {
+    func testChangeStreamResumeToken() throws {
         guard MongoSwiftTestCase.topologyType != .single else {
             print("Skipping test case because of unsupported topology type \(MongoSwiftTestCase.topologyType)")
             return
         }
 
         let client = try MongoClient()
+        let db = client.db(type(of: self).testDatabase)
+        defer { try? db.drop() }
+        let coll = try db.createCollection(self.getCollectionName(suffix: "1"))
+        let changeStream1 = try coll.watch()
+        try coll.insertOne(["x": 1])
+        changeStream1.next()
+        expect(changeStream1.error).to(beNil())
+        // save the current resumeToken and use it as the resumeAfter in a new change stream
+        let resumeAfter = changeStream1.resumeToken
+        let changeStream2 = try coll.watch(options: ChangeStreamOptions(resumeAfter: resumeAfter))
+        // expect this change stream to have its resumeToken set to the resumeAfter
+        expect(changeStream2.resumeToken).to(equal(resumeAfter))
+        expect(changeStream2.error).to(beNil())
+    }
 
-        guard try client.serverVersion() >= ServerVersion(major: 4, minor: 0) else {
-            print("Skipping test case for server version \(try client.serverVersion())")
+    func testChangeStreamResumeTokenError() throws {
```

Comment:
this name doesn't seem to match what the test is checking.

Reply:
I thought this error was related to the resumeToken. I will change it to `testChangeStreamProjectOutIdError` because I see now how it will describe the test better.

Reply:
ahhh, i see. you're right, it does relate to the resume token. 👍to more descriptive name though!